addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComMessage.java

```diff
@@ -46,6 +47,15 @@
     byte[] decodeMessage() throws RFXComException;
 
     /**
+     * Procedure for converting RFXCOM value to openHAB command.
+     *
+     * @param valueSelector
+     *
+     * @return Openhab command.
+     */
+    Command convertToCommand(RFXComValueSelector valueSelector) throws RFXComException;
```

Comment:
How? The first patch in the series simply case the State returned from convertToState() into a Command. This is still the default implementation in RFXComBaseMessage. But Lighting1 (aka X10 and ilk) need to map BRIGHT/DIM to INCREASE/DECREASE and INCREASE/DECREASE are only valid as Commands, not States. So either the protocol special handling goes in the protocol specific message handling or the upper level RFXComHandler needs to access the protocol specific details of messages and know what they mean so they can be handled appropriately. Adding a convertToCommand() method to parallel convertToStatus() seems like a logical and necessary evil, no?